yt/fields/fluid_fields.py

```diff
@@ -195,12 +195,29 @@ def grad_func(axi, ax):
         slice_3dl = slice_3d[:axi] + (sl_left,) + slice_3d[axi+1:]
         slice_3dr = slice_3d[:axi] + (sl_right,) + slice_3d[axi+1:]
         def func(field, data):
-            ds = div_fac * data[ftype, "d%s" % ax]
-            f  = data[grad_field][slice_3dr]/ds[slice_3d]
-            f -= data[grad_field][slice_3dl]/ds[slice_3d]
+            block_reorder = getattr(data, '_block_reorder', 'C')
+            if block_reorder == 'F':
```

Comment:
This causes many tests to fail. It is however required for RAMSES datasets (otherwise the gradients are in the wrong direction).
The issue is that
1. I do not want to add some RAMSES-specific switch here
2. I do not know why only RAMSES requires this reordering among Fortran-ordered datasets.